hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -987,6 +987,9 @@ public Object immediateLoad(String entityName, Object id) throws HibernateExcept
 			event.setResult( null );
 			loadEvent = event;
 		}
+		if ( result instanceof HibernateProxy ) {
+			return ( (HibernateProxy) result ).getHibernateLazyInitializer().getImplementation();
```

Comment:
Doesn't this break the identity guarantee, that entities with the same PK have the same identity, required by JPA? IMO we should trigger lazy loading here, but still return the proxy.

Reply:
it is due to the change in the Initializers logic, now in case of existing proxy they always return the proxy initialised, in 5.x  the entity was loaded and then based on the case a corresponding proxy was searched and initialised.

Reply:
I think the result is the same but not 100% sure